crates/sui-core/src/math.rs

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
```

Comment:
I am aware that @andll suggested introducing our own methods for calculating medians etc, but since we are doing this then I strongly suggest adding tests to ensure that math here works correctly.

Reply:
added